transformer_models.py

```diff
@@ -177,15 +177,17 @@ def create_layers_and_inputs_on_gpu(self):
 class _AssignCache(torch.autograd.Function):
     # Adapted from https://discuss.pytorch.org/t/disable-in-place-correctness-version-check-any-other-workaround/90738/4
     @staticmethod
-    def forward(ctx, cache, value, start, end):
-        ctx.start = start
-        ctx.end = end
-        cache.data[:, start:end, :] = value
+    def forward(ctx, cache, value, index):
+        ctx.index = index
+        cache.data[index] = value
```

Comment:
would this further increase memory fragments?

Reply:
well i don't think so. it's still being done in place right?

Reply:
you are right